src/TextField.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import cx from 'classnames';
+
+/**
+ * @module TextField
+ */
+class TextField extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value,
+			error: props.error
+		};
+		this.handleChange = this.handleChange.bind(this);
+	}
+
+	handleChange(e) {
+		this.setState({ value: e.target.value });
+	}
+
+	render() {
+		const {
+			label,
+			labelClassName,
+			className,
+			children,
+			error,
+			...other
+		} = this.props;
+
+		const classNames = cx(
+			{ 'field--error': this.state.error },
+			className
+		);
+
+		const labelClassNames = cx(
+			{ required : other ? other.required : false },
+			labelClassName
+		);
+
+		const errorElClassNames = cx(
+			'text--error',
+			{ 'display--none': !this.state.error }
```

Comment:
in React, it's _much_ better to just skip rendering entirely for elements that are hidden under certain conditions. In the JSX, use this pattern instead of setting a `classNames` var:

```jsx
{ !this.state.error && <p className='text--error'>{error}</p> }
```